x/proof/keeper/msg_server_submit_proof_test.go

```diff
@@ -1115,101 +1136,120 @@ func createKeypair(
-	supplierKeyUid string,
-	supplierAddr string,
-	requestHeader *sessiontypes.SessionHeader,
-	responseHeader *sessiontypes.SessionHeader,
+	supplierKeyUid, supplierAddr string,
+	reqHeader, resHeader *sessiontypes.SessionHeader,
 	keyRing keyring.Keyring,
 	ringClient crypto.RingClient,
 ) *servicetypes.Relay {
 	t.Helper()
 
-	relay := newEmptyRelay(requestHeader, responseHeader)
-	signRelayRequest(ctx, t, requestHeader.GetApplicationAddress(), keyRing, ringClient, relay)
-	signRelayResponse(t, supplierKeyUid, supplierAddr, keyRing, relay)
+	relay := newEmptyRelay(reqHeader, resHeader)
+	signRelayRequest(ctx, t, relay, reqHeader.GetApplicationAddress(), keyRing, ringClient)
+	signRelayResponse(ctx, t, relay, supplierKeyUid, supplierAddr, keyRing)
 
 	return relay
 }
 
-func newEmptyRelay(
-	requestHeader *sessiontypes.SessionHeader,
-	responseHeader *sessiontypes.SessionHeader,
-) *servicetypes.Relay {
+// newEmptyRelay creates a new relay structure for the given req & res headers
+// WITHOUT any payload or signatures.
+func newEmptyRelay(reqHeader, resHeader *sessiontypes.SessionHeader) *servicetypes.Relay {
 	return &servicetypes.Relay{
 		Req: &servicetypes.RelayRequest{
 			Meta: servicetypes.RelayRequestMetadata{
-				SessionHeader: requestHeader,
-				Signature:     nil, // Signature addded elsewhere.
+				SessionHeader: reqHeader,
+				Signature:     nil, // Signature added elsewhere.
 			},
 			Payload: nil,
 		},
 		Res: &servicetypes.RelayResponse{
 			Meta: servicetypes.RelayResponseMetadata{
-				SessionHeader:     responseHeader,
+				SessionHeader:     resHeader,
 				SupplierSignature: nil, // Signature added elsewhere.
 			},
 			Payload: nil,
 		},
 	}
 }
 
+// TODO_DISCUSS(@red-0ne, @Olshansk): Should this logic be centralized
+// in the relayer package?
+// signRelayRequest signs the relay request (updates relay.Req.Meta.Signature)
+// on behalf of appAddr using the clients provided.
 func signRelayRequest(
 	ctx context.Context,
 	t *testing.T,
+	relay *servicetypes.Relay,
 	appAddr string,
 	keyRing keyring.Keyring,
 	ringClient crypto.RingClient,
-	relay *servicetypes.Relay,
 ) {
 	t.Helper()
 
+	// Retrieve the signing ring associated with the application address.
 	appRing, err := ringClient.GetRingForAddress(ctx, appAddr)
 	require.NoError(t, err)
 
+	// Retrieve the signing key associated with the application address.
 	signingKey := getSigningKeyFromAddress(t,
 		appAddr,
 		keyRing,
 	)
 
+	// Retrieve the signable bytes for the relay request.
 	relayReqSignableBz, err := relay.GetReq().GetSignableBytesHash()
 	require.NoError(t, err)
 
+	// Sign the relay request.
 	signature, err := appRing.Sign(relayReqSignableBz, signingKey)
 	require.NoError(t, err)
 
+	// Serialize the signature.
 	signatureBz, err := signature.Serialize()
 	require.NoError(t, err)
 
+	// Update the relay request signature.
 	relay.Req.Meta.Signature = signatureBz
 }
 
+// TODO_DISCUSS(@red-0ne, @Olshansk): Should this logic be centralized
```

Comment:
You can change this to a `TODO_TECHDEBT`. There's definitely room for factoring this out.